ese-varo/growing
# Feature/user update

## Explain why this change is being made.
- As a registered user, I can access my profile and modify my account information, so I can keep it up to date
## Explain how this is accomplished.
- Having a registered account into the app, when you are logged in you can update your profile information

## How do you manually test this?
- Given a button to edit profile, when the user clicks on it, then the edit profile page is displayed
  - A form with the user account information is displayed
  - The user should be able to edit all the attributes of its account
    - First name
    - Last name
    - Phone
    - Email
    - Password
  - The user clicks on the ‘Update’ button
- Given the user update the account attributes successfully
  - The user should be redirected to the show profile page
  - An alert message should be displayed saying:  “Your account has been updated successfully”
- Given the user has passed the client side validations but can’t successfully update the account information
  - The page should be rerendered with the account information
  - Server side validations feedback should be displayed
    - Existing emails can’t be taken
- Given a form filled with the user account information, when the user enters invalid data on the input fields and clicks on the ‘Update’ button
  - Client side validations for every input field should be performed
  - All fields are required (except new password and password confirmation fields, those are required just if the user want to update the password)

## Screenshots
Show profile view: 
Edit profile view:
